agent/docs/none-platform.md

```diff
@@ -0,0 +1,119 @@
+# None Platform Support
+
+## Configuration Variable
+
+To enable None platform for compact and HA clusters:
+````
+export AGENT_PLATFORM_TYPE=none
+````
+
+### Network Configuration
+
+With None platform, the user is responsible for configuring the DNS entries for
+the Kubernetes API, OpenShift application wildcard, and the names of the control plane
+and compute machines. Users are also responsible for providing a load balancer
+infrastructure for the API and application ingress. 
+
+In the context of dev-scripts, None platform requires these DNS records to be 
+present in the 'ostestbm' network.
+* api.<cluster_name>.<base_domain> 
+* api-int.<cluster_name>.<base_domain> 'new'
+* *.apps.<cluster_name>.<base_domain>. 'new'
+
+All three names point to the load balancer's IP address. 
+
+The DNS records are add to the ostestbm libvirt network by the enable_load_balancer() 
+function in 'agent/05_agent_configure.sh'.
+
+````
+sudo virsh net-dumpxml ostestbm
+<network xmlns:dnsmasq='http://libvirt.org/schemas/network/dnsmasq/1.0'>
+  <name>ostestbm</name>
+  <uuid>f08b75ac-419e-4c0a-8889-45935533f2f6</uuid>
+  <forward mode='nat'>
+    <nat>
+      <port start='1024' end='65535'/>
+    </nat>
+  </forward>
+  <bridge name='ostestbm' stp='on' delay='0'/>
+  <mac address='52:54:00:a4:61:9b'/>
+  <domain name='ostest.test.metalkube.org' localOnly='yes'/>
+  <dns>
+    <forwarder domain='apps.ostest.test.metalkube.org' addr='127.0.0.1'/>
+    <host ip='192.168.111.2'>
+      <hostname>ns1</hostname>
+    </host>
+    <host ip='192.168.111.80'>
+      <hostname>master-0</hostname>
+    </host>
+    <host ip='192.168.111.81'>
+      <hostname>master-1</hostname>
+    </host>
+    <host ip='192.168.111.82'>
+      <hostname>master-2</hostname>
+    </host>
+    <host ip='192.168.111.83'>
+      <hostname>worker-0</hostname>
+    </host>
+    <host ip='192.168.111.84'>
+      <hostname>worker-1</hostname>
+    </host>
+    <host ip='192.168.111.1'>
+      <hostname>api</hostname>
+      <hostname>api-int</hostname>
+      <hostname>*.apps</hostname>
+      <hostname>virthost</hostname>
+    </host>
+  </dns>
+  ** snip **
+  <dnsmasq:options>
+    <dnsmasq:option value='cache-size=0'/>
+  </dnsmasq:options>
+</network>
+````
+
+'haproxy' is the load balancer deployed for enabling the None platform with the agent-based installer. The haproxy service
+is configured and enabled by the enable_load_balancer() function.
+
+'haproxy' runs on the hypervisor host and is accessed by the hosts forming the OpenShift cluster through
+the 'ostestbm' network through that network's .1 or ::1 IP address. Its configuration file path is /etc/haproxy/haproxy.cfg.
```

Comment:
Nit - instead of "through the 'ostestbm' network through that network's..." can just use
"through the ostestbm network's"

Reply:
Thanks Bob. That wasn't pretty.